pkgs/development/coq-modules/ssprove/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib, mkCoqDerivation, coq, version ? null
+, equations
+, mathcomp
+, mathcomp-analysis
+, extructures
+, deriving
+}:
+
+(mkCoqDerivation {
+  pname = "ssprove";
+  owner = "SSProve";
+
+  inherit version;
+  defaultVersion = with lib.versions; lib.switch [coq.coq-version mathcomp.version] [
+    { cases = ["8.19" "2.2.0" ]; out = "0.2.0"; }
+    { cases = ["8.18" "2.1.0" ]; out = "0.2.0"; }
+    # This is the original dependency:
+    # { cases = ["8.17" "1.18.0"]; out = "0.1.0"; }
+    # But it is not loadable. The math-comp nixpkgs configuration
+    # will always only output version 1.18.0 for Coq 8.17.
+    # Hence, the Coq 8.17 and math-comp 1.17.0 must be explicitly set
+    # to load it.
+    # (This version is not on the math-comp CI and hence not checked.)
+    { cases = ["8.17" "1.17.0"]; out = "0.1.0"; }
+  ] null;
+
+  releaseRev = v: "v${v}";
+
+  release."0.2.0".sha256 = "sha256-GDkWH0LUsW165vAUoYC5of9ndr0MbfBtmrPhsJVXi3o=";
+  release."0.1.0".sha256 = "sha256-Yj+k+mBsudi3d6bRVlZLyM4UqQnzAX5tHvxtKoIuNTE=";
+
+  propagatedBuildInputs = [equations
+                           mathcomp
```

Comment:
```suggestion
```

Is all of `mathcomp` really needed?

Reply:
I can narrow this down to `mathcomp-ssreflect` if desired.

Reply:
Yes it matters. Spurious dependencies burn bandwidth, disk space, CI time, etc.

Reply:
Ok, will change it then.